src/router.rs

```diff
@@ -5,52 +5,39 @@ use crate::parser::parse;
 use crate::schema::Schema;
 use crate::semantics::{FieldCounter, Validate};
 use std::collections::{BTreeMap, HashMap};
-use std::ops::Deref;
 use uuid::Uuid;
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
 struct MatcherKey(usize, Uuid);
 
 #[derive(Debug)]
-pub struct Router<'a> {
-    schema: SchemaOwnedOrRef<'a>,
+pub struct Router<S> {
```

Comment:
I'd like to suggest adding a bound here. As `Router<S>` without `AsRef<Schema>` doesn't make any sense here.

```suggestion
pub struct Router<S>
where
    S: AsRef<Schema>
{
```

Reply:
thanks for the suggestion @Oyami-Srk - I had this exact conversation with @lucab who suggested to **remove the bound** because having it would come with a list of downsides that are explained very well in [this SO answer](https://stackoverflow.com/a/66369912/1619736). I don't feel too strongly about going either direction but the points there make the no-bound option feel more appropriate to me. What do you think?

Reply:
Got it! Thanks for the clarification.

Reply:
I think we should leave the bound out of the type definition, which is the way Rust standard library does it as well.